utils/prometheus/src/metrics.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+
+pub use crate::*;
+pub use prometheus::Result;
+/// Gauge type metrics generation function
+pub fn try_create_int_gauge(name: &str, help: &str) -> Result<IntGauge> {
+    let opts = Opts::new(name, help);
+    let gauge = IntGauge::with_opts(opts)?;
+    prometheus::register(Box::new(gauge.clone()))?;
+    Ok(gauge)
+}
+/// histogram type metrics generation function
+pub fn try_create_histogram(name: &str, help: &str) -> Result<Histogram> {
+    let opts = HistogramOpts::new(name, help);
+    let histogram = Histogram::with_opts(opts)?;
+    prometheus::register(Box::new(histogram.clone()))?;
+    Ok(histogram)
+}
+
+///Gauge Metrics a value in injection.
+pub fn set_gauge(gauge: &Result<IntGauge>, value: u64) {
+    if let Ok(gauge) = gauge {
+        gauge.set(value as i64);
+    }
+}
+///histogram Metrics a value in injection.
+pub fn set_histogram(histogram: &Result<Histogram>, value: f64) {
+    if let Ok(histogram) = histogram {
+        histogram.observe(value)
+    }
+}
+//All of the metrics in the prometheus are managed by the lazy_static.
+
+lazy_static! {
+    pub static ref FINALITY_HEIGHT: Result<IntGauge> = try_create_int_gauge(
+        "consensus_finality_block_height_number",
+        "block is finality HEIGHT"
+
+    );
+    pub static ref BEST_HEIGHT: Result<IntGauge> = try_create_int_gauge(
+        "consensus_best_block_height_number",
+        "block is best HEIGHT"
+    );
+
+    pub static ref BLOCK_INTERVAL_SECONDS: Result<Histogram> = try_create_histogram(
+        "consensus_block_interval_seconds",
+        "Time between this and last block(Block.Header.Time) in seconds"
+    );
```

Comment:
Either add a newline here, or remove the above one.

Reply:
complete

Reply:
complete